### ISSUE TITLE
Many SSURGO sites fail when passed into SOILWAT

### ISSUE THREAD

**Issue by Zachary-Kramer** [Burke-Lauenroth-Lab/rSFSW2#55]
**Note:** This contains similar information to issue #46 but addresses a separate problem

**Summary:**
I have been testing my SSURGO extraction with the wrapper/SOILWAT and have quite a few sites that are failing. As far as I can tell, the data is complete, but SOILWAT may dislike some of the numbers (such as huge layer widths). I was wondering if you might have any insight on why these sites are failing. 

**Background:**
* I have 51 sites
* SSURGO has data for 24 of them. STATSGO filled in the other 27.
* 5 out of 24 SSURGO sites succeeded. 27 out of 27 STATSGO sites succeeded. 
* SSURGO will discard layers with empty data, and if the first layer depth is greater than 15cm it will create a dummy layer with a depth of 15, then insert the rest of the layers normally. Additionally, if gravel content is lacking it is filled with a value of 0.01 (this only occurred on 1 of the sites).

**Observations:**
* Reducing the layer widths in soil.layers caused sites to succeed.
* For instance, a site with L1=8cm, L2=46cm, and L3=152cm failed. Changing L2 to 15cm caused the site to succeed. This removed the extreme values for layers 1 and 2, but layer 3 still had the extreme values.
* Layer width is not the only factor, as two sites had the same widths but one failed and the other did not, but it seems to be the main culprit.

**Data:**
Here is a collection of data that's hopefully enough to come to some conclusions. I could not include all of the dumps because they are large, but it might be useful to know that the following sites failed: 9, 13, 18, 19, 20, 21, 22, 24, 25, 26, 27, 28, 30, 35, 36, 37, 38, 41, 42. This should be reproduce-able with the branch I have on rSWSFTools (SSURGODownloadExtract).

[2 example sites.zip](https://github.com/Burke-Lauenroth-Lab/SoilWat_R_Wrapper/files/598517/2.example.sites.zip)
[1_Data_SWInput.zip](https://github.com/Burke-Lauenroth-Lab/SoilWat_R_Wrapper/files/623170/1_Data_SWInput.zip)

I will be moving away from debugging this, because my data extraction task is complete and now it is SOILWAT that dislikes the data provided by SSURGO.

Thanks!

**Comment by dschlaep** [comment]
I cannot replicate the issue.

I loaded both of the provided data files 'last.dump.do_OneSite_X.RData' with X = {9, 28}. Then I ran Rsoilwat 
`sw_exec(inputData = swRunScenariosData[[sc]], weatherList = i_sw_weatherList[[scw]], echo = FALSE, quiet = FALSE)`
This was successful and I didn't encounter an error.

Zach, without the output from the wrapper (ideally with print.debug set to TRUE) I don't know what 'failing' means for you here. Also, the branch 'SSURGODownloadExtract' seems not to up up-to-date with master and this issue may have already been fixed elsewhere.

**Comment by Zachary-Kramer** [comment]
I'm glad to hear that there may not be issues at all. I have to re-create the branch anyways, so when I have time I will do that with the updated master and re-test the input. Failing for me meant I got a dump file for the site in the main directory. The sites "fail" during one of the aggregations, but I will have more information once I get around to testing it again.

**Comment by Zachary-Kramer** [comment]
I have updated SSURGO to run with rSFSW2/rSOILWAT2 and the results are the same. The sites will run through the model (meaning the execution phase) and will generate output in 3_Runs. Almost all sites will fail in the aggregation phase. Some fail at multiple aggregations, some just one, some don't fail at all. 

I have only tested a small number of the SSURGO validation sites and the aggregations that have failed so far are yearlyWaterBalanceFluxes, dailyNRCS_SoilMoistureTemperatureRegimes_Intermediates, and dailyNRCS_SoilMoistureTemperatureRegimes. The errors are very obscure and are ultimately hard to track to the original cause of the issue. As I continue working on this I will open a separate issue for each aggregation that fails. I'm closing this issue because the premise is incorrect (it does not fail in SOILWAT) and the errors I'm running into might too complicated for a general issue.
